components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -79,6 +85,199 @@
         return groups;
     }
 
+    /**
+     * Lists the groups that are created from SCIM
+     *
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException If there is an issue while listing scim groups.
+     */
+    public Set<String> listSCIMGroups(int tenantId, int startIndex, Integer count) throws IdentitySCIMException {
+
+        Set<String> groups = new HashSet<>();
+        String sqlStmt;
+
+        if (startIndex <= 0) {
+            startIndex = 0;
+        } else {
+            startIndex = startIndex - 1;
+        }
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            String dbType = DatabaseCreator.getDatabaseType(connection);
+            if (DB2.equalsIgnoreCase(dbType)) {
+                int initialOffset = startIndex;
+                startIndex = startIndex + count;
+                count = initialOffset + 1;
+                sqlStmt = SQLQueries.LIST_SCIM_GROUPS_WITH_PAGINATED_DB2;
+            } else if (ORACLE.equalsIgnoreCase(dbType)) {
+                count = startIndex + count;
+                sqlStmt = SQLQueries.LIST_SCIM_GROUPS_WITH_PAGINATED_ORACLE;
+            } else if (MSSQL.equalsIgnoreCase(dbType)) {
+                int initialOffset = startIndex;
+                startIndex = count + startIndex;
+                count = initialOffset + 1;
+                sqlStmt = SQLQueries.LIST_SCIM_GROUPS_WITH_PAGINATED_MSSQL;
+            } else {
+                sqlStmt = SQLQueries.LIST_SCIM_GROUPS_WITH_PAGINATION_SQL;
+            }
+
+            try (NamedPreparedStatement statement =
+                         new NamedPreparedStatement(connection, sqlStmt)) {
+                byte increment = 0;
+                statement.setString(++increment, SCIMConstants.CommonSchemaConstants.ID_URI);
```

Comment:
we can't use like that because the index will change according to the availability of the tenantid. This is the way we have handled in other places as well.